### PR TITLE
2945: Distinguish automatic and counselor messages

### PR DESCRIPTION
**Pull request by ztefanie**
### Short description
Added different icons. I tried different styles, check here if you prefer an other option:
image

The backend endpoint always returns false for "automatic_answer". I asked the backend team about this.

### Testing
Open the chat. Change the conditional in ChatMessage.tsx getIcon function to return other icon to see it.

Not ready to merge, because backend endpoint is not working properly. Please add your thoughts on the icons.

Fixes: #2945

---

### CODE REVIEW

File: assets/icons/chatbot-bot.svg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<svg width="400" height="400" viewBox="0 0 400 400" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path d="M399.933 56.72V343.387C399.912 358.413 393.927 372.817 383.293 383.433C372.66 394.048 358.247 400.007 343.223 400H56.6572C41.633 399.993 27.2261 394.02 16.6023 383.395C5.97856 372.769 0.00706526 358.36 0 343.333V56.6667C0.00706526 41.6399 5.97856 27.2306 16.6023 16.6051C27.2261 5.97956 41.633 0.00706643 56.6572 0H343.276C358.309 0.010605 372.721 5.99155 383.346 16.6278C393.971 27.2641 399.937 41.6851 399.933 56.72Z" fill="#FBDA16"/>
```
Comment (steffenkleinle) [inline assets/icons/chatbot-bot.svg:2]: :wrench: At the moment we only plan the chat for integreat, but won't hurt to make the icons configurable already for other build configs.
```suggestion
<path d="M399.933 56.72V343.387C399.912 358.413 393.927 372.817 383.293 383.433C372.66 394.048 358.247 400.007 343.223 400H56.6572C41.633 399.993 27.2261 394.02 16.6023 383.395C5.97856 372.769 0.00706526 358.36 0 343.333V56.6667C0.00706526 41.6399 5.97856 27.2306 16.6023 16.6051C27.2261 5.97956 41.633 0.00706643 56.6572 0H343.276C358.309 0.010605 372.721 5.99155 383.346 16.6278C393.971 27.2641 399.937 41.6851 399.933 56.72Z" fill="currentColor"/>
```

File: assets/icons/chatbot-person.svg [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<svg width="400" height="400" viewBox="0 0 400 400" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path d="M399.933 56.72V343.387C399.912 358.413 393.927 372.817 383.293 383.433C372.66 394.048 358.247 400.007 343.223 400H56.6572C41.633 399.993 27.2261 394.02 16.6023 383.395C5.97856 372.769 0.00706526 358.36 0 343.333V56.6667C0.00706526 41.6399 5.97856 27.2306 16.6023 16.6051C27.2261 5.97956 41.633 0.00706643 56.6572 0H343.276C358.309 0.010605 372.721 5.99155 383.346 16.6278C393.971 27.2641 399.937 41.6851 399.933 56.72Z" fill="#FBDA16"/>
```
Comment (steffenkleinle) [inline assets/icons/chatbot-person.svg:2]:
```suggestion
<path d="M399.933 56.72V343.387C399.912 358.413 393.927 372.817 383.293 383.433C372.66 394.048 358.247 400.007 343.223 400H56.6572C41.633 399.993 27.2261 394.02 16.6023 383.395C5.97856 372.769 0.00706526 358.36 0 343.333V56.6667C0.00706526 41.6399 5.97856 27.2306 16.6023 16.6051C27.2261 5.97956 41.633 0.00706643 56.6572 0H343.276C358.309 0.010605 372.721 5.99155 383.346 16.6278C393.971 27.2641 399.937 41.6851 399.933 56.72Z" fill="currentColor"/>
```

File: assets/icons/chatbot-person.svg [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<svg width="400" height="400" viewBox="0 0 400 400" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (steffenkleinle) [inline assets/icons/chatbot-person.svg:1]: :wrench: I think we decided on the naming `chat`, not `chatbot`, could you please rename the two icons?

File: web/src/components/ChatMessage.tsx [hunk 4]
```diff
@@ -51,15 +52,21 @@ type ChatMessageProps = { message: ChatMessageModel; showIcon: boolean }
 
 const ChatMessage = ({ message, showIcon }: ChatMessageProps): ReactElement => {
   // TODO 2799 Check if Remote content is really needed here or how external links will be delivered via api
```
Comment (steffenkleinle) [inline web/src/components/ChatMessage.tsx:54]: :upside_down_face: Could you perhaps remove the comment? The issue is already closed and I know it's not your thing, but would be nice if you could remove it.

File: web/src/components/ChatMessage.tsx [hunk 5]
```diff
@@ -51,15 +52,21 @@ type ChatMessageProps = { message: ChatMessageModel; showIcon: boolean }
...
-  const { body, userIsAuthor } = message
+  const { body, userIsAuthor, isAutomaticAnswer } = message
+
+  const getIcon = (userIsAuthor: boolean, isAutomaticAnswer: boolean): ReactElement => {
+    if (userIsAuthor) {
+      return <Circle>{t('user')}</Circle>
+    }
+    const icon = isAutomaticAnswer ? ChatbotBot : ChatbotPerson
+    return <Icon src={icon} />
+  }
```
Comment (steffenkleinle) [inline web/src/components/ChatMessage.tsx:65]: :upside_down_face: Optional: Extract to own component (perhaps still in the same file as it is not dependent on any props.

**Review by steffenkleinle — COMMENTED** [review]
Tested on firefox

**Comment by steffenkleinle** [comment]
> Added different icons. I tried different styles, check here if you prefer an other option

I mean thats what the issue was for and probably best to ask UI/UX for this. Ping @nikolahoff, have a look at the currently used icons here: https://github.com/digitalfabrik/integreat-app/pull/2963/files

File: assets/icons/chatbot-bot.svg [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<svg width="400" height="400" viewBox="0 0 400 400" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (steffenkleinle) [inline assets/icons/chatbot-bot.svg:1]: :wrench: Please resize to 24x24px.

**Review by LeandraH — APPROVED** [review]
Nice, thank you! Not tested

**Review by steffenkleinle — COMMENTED** [review]
Well done, thank you. It just seems like the response from the CMS is now wrong, I am getting `automatic_answer: false` for a clearly automatic answer...

**Comment by ztefanie** [comment]
Currently blocked due to backend bug: https://chat.tuerantuer.org/digitalfabrik/pl/h51bd7zb5tfc5cye7q8ngghcey

**Comment by ztefanie** [comment]
ok the backend bug is only on Testumgebung, so this could be merged:

> AFAICT there is nothing blocking right now.
The automatic_answer is available in all relevant API versions. For testumgebung it returns the false value in all(?) cases. I expect that this is related to the Zammad Config. For muenchen on webnext it returns also true where it is expected.

**Review by steffenkleinle — APPROVED** [review]

**Pull request merged**